test/jdk/javax/swing/JTable/JTableRightAlignmentTest.java

```diff
@@ -0,0 +1,217 @@
+                SwingUtilities.invokeAndWait(() -> {
+                    int allColumnWidths = 0;
+                    for (int i = 0; i < table.table.getColumnCount(); i++) {
+                        allColumnWidths += table.table.getTableHeader().getColumnModel()
+                                .getColumn(i)
+                                .getWidth();
+                    }
+                    BufferedImage bufferedImage = robot.createScreenCapture(
+                            new Rectangle(table.table.getLocationOnScreen().x,
+                                    table.table.getLocationOnScreen().y,
+                                    table.table.getWidth() - allColumnWidths,
+                                    table.table.getHeight()));
+                    int expectedRGB = bufferedImage.getRGB(0, 0);
+                    for (int x = 0; x < bufferedImage.getWidth(); x++) {
+                        for (int y = 0; y < bufferedImage.getHeight(); y++) {
+                            if (expectedRGB != bufferedImage.getRGB(x, y)) {
+                                saveImage(bufferedImage, "failureImage.png");
+                                failureString = "Test Failed at <" + x + ", " + y + ">";
+                                break;
+                            }
+                        }
+                    }
+                });
+                if (failureString != null) {
+                    throw new RuntimeException(failureString);
+                } else {
+                    System.out.println("Test Passed!");
+                }
+            } finally {
+                SwingUtilities.invokeAndWait(() -> {
+                    if (frame != null) {
+                        frame.dispose();
+                    }
+                });
+            }
+        }
+    }
+
+    private static void saveImage(BufferedImage image, String fileName) {
+        try {
+            ImageIO.write(image, "png", new File(fileName));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static void setLookAndFeel(UIManager.LookAndFeelInfo laf) {
+        try {
+            UIManager.setLookAndFeel(laf.getClassName());
+        } catch (UnsupportedLookAndFeelException ignored) {
+            System.out.println("Unsupported LAF: " + laf.getClassName());
+        } catch (ClassNotFoundException | InstantiationException
+                 | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+}
+
+class CustomTable {
+    public final static int COL_FIRSTNAME = 0;
+    public final static int COL_LASTNAME = 1;
+    public final static int COL_SALARY = 2;
+
+    static final Class[] classes = {
+            String.class,
+            String.class,
+            Float.class,
+    };
+
+    String[] cols = {
+            "First name",
+            "Last name",
+            "Salary",
+    };
+    List<CustomTable.Record> data = new ArrayList();
```

Comment:
```suggestion
    List<CustomTable.Record> data = new ArrayList<>();
```
The right side still uses raw type.